vaadin-app-layout-flow-parent/vaadin-app-layout-flow/src/main/java/com/vaadin/flow/component/applayout/AppLayout.java

```diff
@@ -51,6 +55,36 @@
 
     private Component content;
 
+    private AppLayoutI18n i18n;
+
+    /**
+     * Gets the internationalization object previously set for this component.
+     * <p>
+     * Note: updating the object content that is gotten from this method will
+     * not update the lang on the component if not set back using
+     * {@link AppLayout#setI18n(AppLayoutI18n)}
+     *
+     * @return the i18n object. It will be <code>null</code>, if the i18n
+     *         properties are not set.
+     */
+    public AppLayoutI18n getI18n() {
+        return i18n;
+    }
+
+    /**
+     * Sets the internationalization properties for this component.
+     *
+     * @param i18n
+     *            the internationalized properties, not <code>null</code>
+     */
+    public void setI18n(AppLayoutI18n i18n) {
+        Objects.requireNonNull(i18n,
+                "The I18N properties object should not be null");
+        this.i18n = i18n;
+        JsonObject i18nObject = (JsonObject) JsonSerializer.toJson(i18n);
+        getElement().setPropertyJson("i18n", i18nObject);
```

Comment:
I found that most of the Flow components supporting i18n don't extend the i18n property, but rather override it entirely with `setPropertyJson`. Like you can see in Avatar: 

https://github.com/vaadin/flow-components/blob/29dffc9926fe8781eeaa47ecc26e44e4fb4f836b/vaadin-avatar-flow-parent/vaadin-avatar-flow/src/main/java/com/vaadin/flow/component/avatar/Avatar.java#L138-L144

There is only MenuBar that stands out from the rest as it has a special logic based on `Object.assign` JavaScript API:

https://github.com/vaadin/flow-components/blob/2a1dd622a1530af1daf334fbeaf71efb414d8dfe/vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java#L305-L309

After all, I decided to go with the most common solution that is based on `setPropertyJson`. Any other considerations?

Reply:
Discussed it internally with @sissbruecker: Upload and DatePicker are also on the list of components extending the i18n property. If we do not support extending at the beginning, it will likely emerge later like it was with Upload and DatePicker. So it makes sense to use the `Object.assign` solution proposed in the MenuBar PR.